src/vm/dllimport.cpp

```diff
@@ -6042,20 +6052,26 @@ HMODULE NDirect::LoadFromNativeDllSearchDirectories(AppDomain* pDomain, LPCWSTR
     return hmod;
 }
 
-#ifdef FEATURE_PAL
 static void DetermineLibNameVariations(const char* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
 {
     if (libNameIsRelativePath)
     {
+        bool containsSuffix;
+#ifdef FEATURE_PAL
         // We check if the suffix is contained in the name, because on Linux it is common to append
         // a version number to the library name (e.g. 'libicuuc.so.57').
-        bool containsSuffix;
         SString::CIterator it = libName.Begin();
-        if (libName.FindASCII(it, PAL_SHLIB_SUFFIX))
+        if (libName.FindASCII(it, PLATFORM_SHARED_LIB_SUFFIX_A))
         {
-            it += strlen(PAL_SHLIB_SUFFIX);
+            it += (int)strlen(PLATFORM_SHARED_LIB_SUFFIX_A);
             containsSuffix = it == libName.End() || *it == (WCHAR)'.';
         }
+#else
+        if (libName.EndsWithCaseInsensitive(PLATFORM_SHARED_LIB_SUFFIX_W) || libName.EndsWithCaseInsensitive(WINDOWS_SHARED_EXE_SUFFIX_W))
```

Comment:
I do not understand why we are special casing `.exe` suffix here. `.exe` suffix is non-standard suffix for shared libraries. If I want to, I can use `.dat` suffix for my .dlls and LoadLibrary will load them too.

Reply:
The only reason is to avoid to load/search .exe.dll file. if it is possible that someone name his file as a.exe.dll, then we can remove this special case.

Reply:
I am more worried about that cases that used to work fine before this change, but do not work anymore after this change. Consider:

```csharp
[DllImport("mydll.dat")]
extern static int sum(int a, int b);

static void Main() {
    Console.WriteLine(sum(1,2).ToString());
}
```

And `mydll.dat` that is renamed `mydll.dll` to `mydll.dat`. It works fine before this change (`mydll.dat` is passed to LoadLibrary), but it does not work after this change (`mydll.dat.dll` is passed to LoadLibrary).

Reply:
Okay. 
`
            static const char* const SuffixFirst[] =
             {
                 "%.0s%s%s", // name+suffix
#ifdef FEATURE_PAL
                 "%s%s%s",   // prefix+name+suffix
#endif
                 "%.0s%s",   // name
#ifdef FEATURE_PAL
                 "%s%s%.0s"  // prefix+name
#endif
             };
'

Reply:
I can check // name first, then //name + suffix to be more comp with current behavior.